### PR TITLE
VTX-3411: skip empty partitions in shuffle reader

### PR DESCRIPTION
**Pull request by fsdvh**
Skip empty partitions in the shuffle reader

### CODE REVIEW

File: ballista/core/src/execution_plans/shuffle_reader.rs [hunk 1]
```diff
@@ -184,7 +184,11 @@ impl ExecutionPlan for ShuffleReaderExec {
         // Sort partitions for evenly send fetching partition requests to avoid hot executors within one task
         let mut partition_locations: Vec<PartitionLocation> = partition_locations
             .into_values()
-            .flat_map(|ps| ps.into_iter().enumerate())
+            .flat_map(|ps| {
```
Comment (thinkharderdev) [inline ballista/core/src/execution_plans/shuffle_reader.rs:187]: We just need to test this and make sure it works correctly when all partitions are empty

Comment (fsdvh) [reply ballista/core/src/execution_plans/shuffle_reader.rs:187]: I plan to do it on staging

**Review by mpurins-coralogix — APPROVED** [review]

**Review by thinkharderdev — APPROVED** [review]

**Review by ch-sc — APPROVED** [review]
LGTM

Still, I'm wondering how did you create empty partitions for testing?

**Comment by mpurins-coralogix** [comment]
> Still, I'm wondering how did you create empty partitions for testing?

I think that can be easily achieved by making sure you have two stage query with all the data filtered out.

**Comment by fsdvh** [comment]
> LGTM
> 
> Still, I'm wondering how did you create empty partitions for testing?

Yeah, so I made a few things:

Executed such query:
```
filter $d.log_obj.message == 'submit_query' | extract $d.log_obj.target into $d using regexp(e=/(?<first>([a-z]+))::(?<second>[a-z]+)/) | filter $d.first == 'bapi'
```

Checked logs:
Screenshot 2024-01-29 at 15 18 11

Query doesn't fail ✅

**Comment by ch-sc** [comment]
Nice, thanks @fsdvh!
